porter/v1/cluster_control_plane.proto

```diff
@@ -435,6 +435,20 @@ message DeleteAddonRequest {
 
 message DeleteAddonResponse {}
 
+message AddonPreflightCheckRequest {
+  int64 project_id = 1;
+  int64 cluster_id = 2;
+  // deployment_target_identifier is the object that identifies the deployment target for the addon
+  // If cluster_id is specified and deployment_target_identifier is not, then the default deployment target for the provided cluster will be used
+  DeploymentTargetIdentifier deployment_target_identifier = 3;
+  AddonType addon_type = 4;
```

Comment:
this should take in the entire addon config. example is if we ever want to expose the instance type of the deepgram addon to client, that will require different quotas